src/main/java/io/github/sashirestela/cleverclient/CleverClient.java

```diff
@@ -22,17 +22,23 @@
  */
 @Getter
 public class CleverClient {
-    private static Logger logger = LoggerFactory.getLogger(CleverClient.class);
+    private static final Logger logger = LoggerFactory.getLogger(CleverClient.class);
 
-    private String urlBase;
-    private List<String> headers;
-    private HttpClient httpClient;
-    private HttpProcessor httpProcessor;
+    private final String baseUrl;
+    @Deprecated
+    private final String urlBase = ""; // never need to use this deprecated
```

Comment:
Please, move this comment to the constructor's Javadoc.

Do we need to initialize it to an empty string?

Reply:
Yes. If we don't initialize we get an error that it is not initialized. We can initialized in the constructor, but the whole point is that this field is never used. Its only serves as an anchor for @Builder to create the backwards-compatible `.urlBase()` method of the builder.